test/remote/gateways/remote_stripe_payment_intents_test.rb

```diff
@@ -64,13 +88,80 @@ def test_successful_purchase
     assert purchase.params.dig('charges', 'data')[0]['captured']
   end
 
-  def test_unsuccessful_purchase_apple_pay
+  def test_unsuccessful_purchase_google_pay_with_invalid_card_number
+    options = {
+      currency: 'GBP'
+    }
+
+    @google_pay.number = '378282246310000'
+    purchase = @gateway.purchase(@amount, @google_pay, options)
+    assert_equal 'Your card number is incorrect.', purchase.message
+    assert_false purchase.success?
+  end
+
+  def test_unsuccessful_purchase_google_pay_without_cryptogram
+    options = {
+      currency: 'GBP'
+    }
+    @google_pay.payment_cryptogram = ''
+    purchase = @gateway.purchase(@amount, @google_pay, options)
+    assert_equal "Cards using 'tokenization_method=android_pay' require the 'cryptogram' field to be set.", purchase.message
+    assert_false purchase.success?
+  end
+
+  def test_unsuccessful_purchase_google_pay_without_month
+    options = {
+      currency: 'GBP'
+    }
+    @google_pay.month = ''
+    purchase = @gateway.purchase(@amount, @google_pay, options)
+    assert_equal 'Missing required param: card[exp_month].', purchase.message
+    assert_false purchase.success?
+  end
+
+  def test_successful_authorize_with_google_pay
+    options = {
+      currency: 'GBP'
+    }
+
+    auth = @gateway.authorize(@amount, @google_pay, options)
+    assert auth.success?
+  end
+
+  def test_successful_purchase_with_apple_pay
+    options = {
+      currency: 'GBP'
+    }
+
+    purchase = @gateway.purchase(@amount, @apple_pay, options)
+    assert purchase.success?
+  end
+
+  def test_successful_authorize_and_void_google_pay
```

Comment:
💬  Comment:

I'm not sure if is worth checking this kind of post transaction test, I mean, no matter if the initial payment was performed with Google pay or Apple pay, the transaction was charged to the underlying real credit card, to so void and refund should work in the same as if the payment was made directly with a Credit Card.